sky/packages/sky/lib/src/gestures/tap.dart

```diff
@@ -2,37 +2,174 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:async';
+import 'dart:ui' as ui;
+
 import 'arena.dart';
+import 'constants.dart';
 import 'events.dart';
+import 'pointer_router.dart';
 import 'recognizer.dart';
 
 typedef void GestureTapCallback();
 
-class TapGestureRecognizer extends PrimaryPointerGestureRecognizer {
-  TapGestureRecognizer({ PointerRouter router, this.onTap })
-    : super(router: router);
+enum TapResolution {
+  tap,
+  cancel
+}
+
+/// TapTracker helps track individual tap sequences as part of a
+/// larger gesture.
+class TapTracker {
+
+  TapTracker({ PointerInputEvent event, this.entry })
+  : pointer = event.pointer,
+    initialPosition = event.position,
+    isTrackingPointer = false {
+      assert(event.type == 'pointerdown');
+    }
```

Comment:
lines 26-28 should be intended two more spaces and lines 29-30 should be indented two fewer spaces.  :)